backend/test/integration/healthz_api_test.go

```diff
@@ -43,10 +44,27 @@ func (s *HealthzApiTest) SetupTest() {
 			glog.Exitf("Failed to initialize test. Error: %v", err)
 		}
 	}
+
 	s.namespace = *namespace
-	clientConfig := test.GetClientConfig(*namespace)
+
+	var newHealthzClient func() (*api_server.HealthzClient, error)
+
+	if *isKubeflowMode {
+		s.resourceNamespace = *resourceNamespace
+
+		newHealthzClient = func() (*api_server.HealthzClient, error) {
+			return api_server.NewKubeflowInClusterHealthzClient(s.namespace, *isDebugMode)
```

Comment:
Just to confirm, what's the difference between `s.resourceNamespace` and `s.namespace`? Should we use the former in full Kubeflow mode?

Reply:
s.namespace is the namespace where kfp is installed (i.e. kubeflow). s.resourceNamespace is where the KFP resources (e.g. experiments) will be created (i.e. kf-conformance).

The API client is created against s.namespace so it can talk to ml-pipeline server.